language/diem-tools/prefetched-transaction-replay/src/lib.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashSet;
+use std::convert::TryFrom;
+use std::path::PathBuf;
+
+use anyhow::{anyhow, bail, format_err, Result};
+use diem_client::{
+    views::AccountStateWithProofView, AccountAddress, BlockingClient, MethodRequest, MethodResponse,
+};
+use diem_read_write_set::ReadWriteSetAnalysis;
+use diem_state_view::StateView;
+use diem_transaction_replay::DiemDebugger;
+use diem_types::access_path::AccessPath;
+use diem_types::transaction::TransactionOutput;
+use diem_types::{
+    account_state::AccountState, account_state_blob::AccountStateBlob, transaction::Transaction,
+};
+use diem_vm::{DiemVM, VMExecutor};
+use move_binary_format::CompiledModule;
+use move_cli::sandbox::utils::{
+    mode::{Mode, ModeType},
+    on_disk_state_view::OnDiskStateView,
+};
+use move_core_types::language_storage::ResourceKey;
+
+pub struct OnDiskStateViewReader {
+    view: OnDiskStateView,
+}
+
+struct AccountAddressWithState {
+    address: AccountAddress,
+    state: AccountState,
+}
+
+pub struct PrefetchTransactionReplayerConfig {
+    build_dir: PathBuf,
+    storage_dir: PathBuf,
+    rpc_url: String,
+    module_mode_type: ModeType,
```

Comment:
I think here, there's no need to make the mode type configurable, we always want "bare"